File: src/test/resources/js/tests/rules/rectangle.js
const NS = "http://datashapes.org/js/tests/rules/rectangle.test#";

function computeArea($this) {
    let width = getProperty($this, "width");
    let height = getProperty($this, "height");
    let area = TermFactory.literal(width.getLex() * height.getLex(), width.getDatatype());
    let areaProperty = TermFactory.namedNode(NS + "area");
    return [ [$this, areaProperty, area] ];
}

function getProperty($this, name) {
    let it = $data.find($this, TermFactory.namedNode(NS + name), null);
    let result = it.next().getObject();
    it.close();
    return result;
}

File: src/test/resources/py/tests/rules/rectangle.py
ns = "http://datashapes.org/py/tests/rules/rectangle.test#"


def compute_area(_this):
    width = get_property(_this, "width")
    height = get_property(_this, "height")
    area = py_tf.literal(int(width.getLex()) * int(height.getLex()), width.getDatatype())
    area_property = py_tf.namedNode(ns + "area")
    return [ [_this, area_property, area] ]


def get_property(_this, name):
    it = _data.find(_this, py_tf.namedNode(ns + name), None)
    result = it.next().getObject()
    it.close()
    return result
